Tools/TestWebKitAPI/Tests/WebKitCocoa/SiteIsolation.mm

```diff
@@ -4369,6 +4402,214 @@ HTTPServer server({
+    ] completion:^(NSArray<NSError *> *errors) {
+        EXPECT_EQ(errors, nil);
+        done = true;
+    }];
+    Util::run(&done);
+
+    [webView evaluateJavaScript:@"getTextContent()" completionHandler:nil];
+    Util::run(&receivedMessage2);
+
+    [webView evaluateJavaScript:@"getIframeTextContent()" completionHandler:nil];
+    Util::run(&receivedMessage3);
+}
+
+TEST(SiteIsolation, CompleteTextManipulationFailsInSomeFrame)
+{
+    static constexpr auto mainFrameBytes = R"TESTRESOURCE(
+    <div>mainframe content</div>
+    <script>
+        function removeIframe() {
+            let element = document.getElementById('iframe');
+            element.parentNode.removeChild(element);
+        }
+        var messageCount = 0;
+        addEventListener('message', () => {
+            if (++messageCount == 2)
+                window.webkit.messageHandlers.testHandler.postMessage('loaded');
+        }, false);
+    </script>
+    <iframe id='iframe' src='https://apple.com/apple'></iframe>
+    <iframe src='https://apple.com/apple'></iframe>
+    )TESTRESOURCE"_s;
+
+    static constexpr auto iframeBytes = R"TESTRESOURCE(
+    <div>iframe content</div>
+    <script>
+        parent.postMessage('loaded', '*');
+    </script>
+    )TESTRESOURCE"_s;
+
+    HTTPServer server({
+        { "/example"_s, { mainFrameBytes } },
+        { "/apple"_s, { iframeBytes } },
+    }, HTTPServer::Protocol::HttpsProxy);
+
+    bool receivedMessage = false;
+    auto webViewAndDelegates = makeWebViewAndDelegates(server);
+    [webViewAndDelegates.messageHandler addMessage:@"loaded" withHandler:[&]() {
+        receivedMessage = true;
+    }];
+    RetainPtr webView = webViewAndDelegates.webView;
+    RetainPtr textManipulationDelegate = adoptNS([[SiteIsolationTextManipulationDelegate alloc] init]);
+    [webView _setTextManipulationDelegate:textManipulationDelegate.get()];
+    RetainPtr manipulationConfiguration = adoptNS([[_WKTextManipulationConfiguration alloc] init]);
+    manipulationConfiguration.get().includeSubframes = YES;
+
+    [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"https://example.com/example"]]];
+    Util::run(&receivedMessage);
+
+    [webView _startTextManipulationsWithConfiguration:manipulationConfiguration.get() completion:^{ }];
+    while ([textManipulationDelegate items].count < 3)
+        Util::spinRunLoop();
+
+    RetainPtr items = [textManipulationDelegate items];
+    auto sortFunction = ^(_WKTextManipulationItem *item1, _WKTextManipulationItem *item2) {
+        auto value1 = [NSNumber numberWithBool:item1.isSubframe];
+        auto value2 = [NSNumber numberWithBool:item2.isSubframe];
+        return [value1 compare:value2];
+    };
+
+    RetainPtr sortedItems = [items.get() sortedArrayUsingComparator:sortFunction];
+    EXPECT_EQ(items.get().count, 3UL);
+    auto item1 = [sortedItems objectAtIndex:0];
+    auto item2 = [sortedItems objectAtIndex:1];
+    auto item3 = [sortedItems objectAtIndex:1];
```

Comment:
Isn't this supposed to be at objectAtIndex:2? We should probably use a distinct content for frame 2 and frame 3 to avoid this kind of error.